sysdeps/nacl/ioctl.c

```diff
@@ -0,0 +1,37 @@
+#include <errno.h>
+#include <stdarg.h>
+#include <sys/ioctl.h>
+
+#include <irt_syscalls.h>
+
+int __ioctl (int fd, unsigned long request, ...) {
```

Comment:
I don't think the use of a variadic function is either necessary or helpful... I think just having a 3rd void* argument would be better? unless there is something I am missing.

Reply:
I wanted to keep the implementation the same as the POSIX standard as the Linux man-pages specified the function to be called this way.

The function can also be called without any parameters so I am not sure if putting the third argument there is the best approach.

Reply:
I think you're probably correct, the signature doesn't matter *that* much considering this is a library function but overall I think you're right that using variadic arguments is less confusing than the alternative here (even though variadic arguments in C are a total mess).